mainwindow.cpp

```diff
@@ -577,78 +513,96 @@ QStringList MainWindow::getAttackParameters()
     } else {
         // In case we do not have explicit format we erase remembered
         // key.
-        m_format = "";
+        m_format.clear();
     }
+    m_settings.setValue("formatJohn", m_format);
+    m_settings.setValue("formatUI", m_ui->comboBox_Format->currentText());
+    
     // Modes
     QWidget* selectedMode = m_ui->attackModeTabWidget->currentWidget();
     if (selectedMode == m_ui->defaultModeTab) {
         // Default behaviour - no modes
         // There are no options here.
+        m_settings.setValue("mode", "default");
     } else if (selectedMode == m_ui->singleModeTab) {
         // "Single crack" mode
         parameters << "--single";
+        m_settings.setValue("mode", "single");
         // External mode, filter
-        if (m_ui->checkBox_SingleCrackModeExternalName->isChecked())
+        if (m_ui->checkBox_SingleCrackModeExternalName->isChecked()) {
```

Comment:
I used to prefer open curly braces on a new line, but in the TODO PR, you told me to read again the qt coding style website which says on the same line so I started modifying my code. Aleksey's code was already following the "on the same line" principle.